openlibrary/plugins/upstream/utils.py

```diff
@@ -734,6 +734,25 @@ def canonical_url(self):
         """
         readable_path = web.ctx.get('readable_path', web.ctx.path) or ''
         query = web.ctx.query or ''
+
+        if len(query) > 0:
+            queries_to_exclude = ['sort', 'mode', 'v', 'type', 'debug']
+            parameters = query.split('&')
+            pattern = re.compile('.*?(?==)')
+            params_to_remove = []
+
+            for param in parameters:
+                p = pattern.match(param).group().replace('?', '')
+                if p in queries_to_exclude:
+                    params_to_remove.append(param)
+
+            for p in params_to_remove:
+                parameters.remove(p)
```

Comment:
This could be simplified as a list comprehension which includes only the elements that pass your filter, but since it's all going away, we won't worry about that. :)